pygmt/clib/session.py

```diff
@@ -1678,3 +1680,74 @@ def extract_region(self):
         if status != 0:
             raise GMTCLibError("Failed to extract region from current figure.")
         return wesn
+
+    def read_virtualfile(self, vfname, kind=None):
```

Comment:
Probably need to add some unit tests to `test_clib...` for this. Since that `test_clib.py` file is getting long, should we split the virtualfile related tests into a `test_clib_virtualfile.py`? I was thinking of doing this before continuing work on #2774.

Reply:
> Probably need to add some unit tests to `test_clib...` for this. Since that `test_clib.py` file is getting long, should we split the virtualfile related tests into a `test_clib_virtualfile.py`? I was thinking of doing this before continuing work on #2774.

Sounds good to me.

Reply:
I'll open a PR for it.

Reply:
Done at #2784.